library/src/blas1/rocblas_swap.hpp

```diff
@@ -13,35 +13,66 @@ __forceinline__ __device__ __host__ void rocblas_swap_vals(T* x, T* y)
     *x    = tmp;
 }
 
-template <typename T>
-__global__ void rocblas_swap_kernel(rocblas_int n, T* x, rocblas_int incx, T* y, rocblas_int incy)
+template <typename U>
```

Comment:
Is there a reason this is `U` instead of `T`? Not that it matters haha

Reply:
Well the original reason was 2 template parameters, probably should have a convention that T is a base type and anything more complex has some hints in the name, as I had used XPtr for another case.  This style I copied from another existing merged template but as you can see using the pre-existing pattern generated a lot of comments.